dotnetv3/EventBridge/Scenarios/EventBridgeScenario.cs

```diff
@@ -0,0 +1,584 @@
+        Console.WriteLine($"\tAdded event rule {eventRuleName} with SNS target {topicName} for bucket {testBucketName}.");
+
+        Console.WriteLine(new string('-', 80));
+    }
+
+    /// <summary>
+    /// List the event rules on the default event bus.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    private static async Task ListEventRules()
+    {
+        Console.WriteLine(new string('-', 80));
+        Console.WriteLine("Current event rules:");
+
+        var rules = await _eventBridgeWrapper.ListAllRulesForEventBus();
+        rules.ForEach(r => Console.WriteLine($"\tRule: {r.Name} Description: {r.Description} State: {r.State}"));
+
+        Console.WriteLine(new string('-', 80));
+    }
+
+    /// <summary>
+    /// Update the event target to use a transform.
+    /// </summary>
+    /// <param name="topicArn">The Amazon SNS topic ARN target to update.</param>
+    /// <returns>Async task.</returns>
+    private static async Task UpdateSnsEventRule(string topicArn)
+    {
+        Console.WriteLine(new string('-', 80));
+        Console.WriteLine("Let's update the event target with a transform.");
+
+        var eventRuleName = _configuration["eventRuleName"];
+        var testBucketName = _configuration["testBucketName"];
+
+        await _eventBridgeWrapper.UpdateS3UploadRuleTargetWithTransform(eventRuleName, topicArn);
+        Console.WriteLine($"\tUpdated event rule {eventRuleName} with SNS target {topicArn} for bucket {testBucketName}.");
+
+        Console.WriteLine(new string('-', 80));
+    }
+
+    /// <summary>
+    /// Update the rule to use a custom event pattern.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    private static async Task UpdateToCustomRule(string topicArn)
+    {
+        Console.WriteLine(new string('-', 80));
+        Console.WriteLine("Updating the event pattern to be triggered by a custom event instead.");
+
+        var eventRuleName = _configuration["eventRuleName"];
+
+        await _eventBridgeWrapper.UpdateCustomEventPattern(eventRuleName);
+
+        Console.WriteLine($"\tUpdated event rule {eventRuleName} to custom pattern.");
+
+        await _eventBridgeWrapper.UpdateCustomRuleTargetWithTransform(eventRuleName,
+            topicArn);
+
+        Console.WriteLine($"\tUpdated event target {topicArn}.");
+
+        Console.WriteLine(new string('-', 80));
+    }
+
+    /// <summary>
+    /// Send rule events for a custom rule using the user's email address.
+    /// </summary>
+    /// <param name="email">The email address to include.</param>
+    /// <returns>Async task.</returns>
+    private static async Task TriggerCustomRule(string email)
+    {
+        Console.WriteLine(new string('-', 80));
+        Console.WriteLine("Sending an event to trigger the rule.");
+
+        await _eventBridgeWrapper.PutCustomEmailEvent(email);
+
+        Console.WriteLine($"\tEvents have been sent. Press enter to continue.");
+        Console.ReadLine();
```

Comment:
Will this even trigger an email? If so, you could mention something like "check your email."

Reply:
I did add some info about when emails will be sent. IF they don't subscribe, they won't get them, so I tried to make it general, like "a subscription email will be sent".